### PR TITLE
samples: usb: add common USB device instantiation and initialization

### PR DESCRIPTION
**Pull request by jfischer-no**
There is common code for the new USB device support in the samples to
instantiate USB device, descriptors, configuration and initialize USB
device. Add common code that can be used for the simple usecase with a
single USB device and single configuration.

### CODE REVIEW

File: samples/subsys/usb/common/sample_usbd.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Comment (tmon-nordic) [inline samples/subsys/usb/common/sample_usbd.h:6]: This file is missing include guards.

File: samples/subsys/usb/cdc_acm/Kconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# Source common USB sample options used to initialize new experimental USB
+# device stack. You do not need to source this Kconfig file in your custom
```
Comment (tmon-nordic) [inline samples/subsys/usb/cdc_acm/Kconfig:5]: Here you state that `not need` but in the Kconfig file you state that it `cannot` be used by application. The two comments should be coherent.

**Review by tmon-nordic — CHANGES_REQUESTED** [review]
Just some minor omissions, otherwise fine.

File: samples/subsys/usb/cdc_acm/Kconfig (hunk 2 above)
Comment (jfischer-no) [reply samples/subsys/usb/cdc_acm/Kconfig]: Changed it, but I am not sure if it is less confusing.

File: samples/subsys/usb/common/sample_usbd.h (hunk 1 above)
Comment (jfischer-no) [reply samples/subsys/usb/common/sample_usbd.h:6]: Then it is more pain than just guarding.

File: samples/subsys/usb/cdc_acm/Kconfig (hunk 2 above)
Comment (jfischer-no) [reply samples/subsys/usb/cdc_acm/Kconfig]: Should be fixed now.

**Review by tmon-nordic — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

File: samples/subsys/usb/console/CMakeLists.txt [hunk 3]
```diff
@@ -4,5 +4,13 @@ cmake_minimum_required(VERSION 3.20.0)
...
+set(SAMPLE_USBD_DIR ${ZEPHYR_BASE}/samples/subsys/usb/common)
 FILE(GLOB app_sources src/*.c)
+
+target_include_directories(app PRIVATE ${SAMPLE_USBD_DIR})
+
 target_sources(app PRIVATE ${app_sources})
+
+target_sources_ifdef(CONFIG_USB_DEVICE_STACK_NEXT app PRIVATE
+	${SAMPLE_USBD_DIR}/sample_usbd_init.c
+)
```
[the remark this answers is not in the document]
Comment (jfischer-no) [reply samples/subsys/usb/console/CMakeLists.txt:16]: I tried and it was not much better, but maybe it was just my cmake incompetence. target_include_directories() needs to be here anyway. Maybe @nordicjm knows.

[the remark this answers is not in the document]
Comment (nordicjm) [reply samples/subsys/usb/console/CMakeLists.txt:16]: the `target_sources_ifdef` part could be moved to a common file, something like include ../../usb_samples.cmake

**Review by fabiobaltieri — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**